### PR TITLE
Send visitor mode and visitor count in conference properties.

### PR DESCRIPTION
**Pull request by JonathanLennox**

### CODE REVIEW

**Comment by JonathanLennox** [comment]
Note: depends on https://github.com/jitsi/jitsi-xmpp-extensions/pull/89

File: jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java [hunk 1]
```diff
@@ -1519,6 +1553,60 @@ private void rescheduleSingleParticipantTimeout()
...
+            Instant now = Instant.now();
+            if (sendNumVisitorsTask != null)
+            {
+                return;
+            }
+            if (visitorsLastUpdated != null &&
+                Duration.between(visitorsLastUpdated, now).
+                    compareTo(VisitorsConfig.config.getNotificationInterval()) < 0)
+            {
+                sendNumVisitorsTask = TaskPools.getIoPool().submit(this::sendNumVisitors);
```
Comment (bgrozev) [inline jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java:1587]: Does this need to be scheduled for later?

Comment (JonathanLennox) [reply jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java:1587]: 🤦

File: jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java [hunk 2]
```diff
@@ -106,6 +107,17 @@
...
 
+    /**
+     * A stat number of conference participants with a visitor muc role.
+     */
+    private RateLimitedStat visitorCount = new RateLimitedStat(VisitorsConfig.config.getNotificationInterval(),
+        (numVisitors) -> {
+            setConferenceProperty(
+                ConferenceProperties.KEY_VISITOR_COUNT,
+                Integer.toString(numVisitors));
+            return null;
```
Comment (bgrozev) [inline jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java:118]: Hm, if the kotlin type is `(Int) -> Unit` should this return `Unit.INSTANCE`?

Comment (bgrozev) [reply jicofo/src/main/java/org/jitsi/jicofo/conference/JitsiMeetConferenceImpl.java:118]: Nevermind, it gets converted to Unit _somewhere_

**Review by bgrozev — APPROVED** [review]

**Review by bgrozev — APPROVED** [review]

**Pull request merged**